pkg/apis/eksctl.io/v1alpha5/types.go

```diff
@@ -1237,6 +1235,9 @@ type NodeGroupBase struct {
 	// Applied to the EKS Nodegroup resource and to the EC2 instances (managed)
 	// +optional
 	Tags map[string]string `json:"tags,omitempty"`
+	// Taints taints to apply to the nodegroup
+	// +optional
+	Taints map[string]string `json:"taints,omitempty"`
```

Comment:
This map is a map of `taintKey` and `taintValue:taintEffect`, right? This will cause override existing taint in the map. Consider this list of taints: `[k=v:NoSchedule,k=v:NoExecute]`, now when you convert this list into the above map, it will have only one element: `{"k":"v:NoExecute"}`. First it will be `{"k":"v:NoSchedule"}` and as soon the 2nd taint is processed, the key value pair will be overwritten!

The correct transformation is this map needs to a map where the key is "taintKey:taintEffect" and value is "taintValue".

Reply:
Good point. Unmanaged nodegroups use the same structure, so I was trying to be consistent. 

I have changed the structure to be a slice of this type:
```
{
	Key    string             `json:"key,omitempty"`
	Value  string             `json:"value,omitempty"`
	Effect corev1.TaintEffect `json:"effect,omitempty"`
}
```